pkg/ebpf/events_pipeline.go

```diff
@@ -262,7 +262,7 @@ func (t *Tracee) decodeEvents(outerCtx context.Context, sourceChan chan []byte)
 				Syscall:             syscall,
 			}
 
-			if !t.shouldProcessEvent(&evt) {
+			if !t.shouldEmitEvent(&evt) {
```

Comment:
emit only happens at the last stage of the pipeline.
I think that removing this wrapper and calling it directly like you did was better, but if keeping the wrapper consider renaming it.